site/src/api/queries/users.ts

```diff
@@ -1,11 +1,15 @@
-import { QueryClient } from "@tanstack/react-query";
+import { QueryClient, QueryOptions } from "@tanstack/react-query";
 import * as API from "api/api";
-import { UpdateUserPasswordRequest, UsersRequest } from "api/typesGenerated";
+import {
+  GetUsersResponse,
+  UpdateUserPasswordRequest,
+  UsersRequest,
+} from "api/typesGenerated";
 
-export const users = (req: UsersRequest) => {
+export const users = (req: UsersRequest): QueryOptions<GetUsersResponse> => {
   return {
     queryKey: ["users", req],
-    queryFn: () => API.getUsers(req),
+    queryFn: ({ signal }) => API.getUsers(req, signal),
```

Comment:
Just making sure – are there any specific places you expect the signal to be called, or is it just for general unmounting?

Reply:
It is a good question, in a perfect scenario all our API functions would support it but it needs some thinking and refactoring. For now, I'm adding it here because it uses debounce and often needs to be canceled.